File: src/backend/CareFirstClinic.API/CareFirstClinic.API/Models/Schedule.cs
using CareFirstClinic.API.Models;

public class Schedule
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid DoctorId { get; set; }

    public DateTime WorkDate { get; set; }

    public TimeSpan StartTime { get; set; }

    public TimeSpan EndTime { get; set; }

    public int SlotDurationMinutes { get; set; }

    public int TotalSlots { get; set; }
    public int AvailableSlots { get; set; }

    public bool IsAvailable { get; set; } = true;

    public string? Note { get; set; }

    // Navigation
    public Doctor? Doctor { get; set; }
    public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
}

File: src/frontend/src/types/schedule.ts
export interface TimeSlot {
  id: string;
  startTime: string;
  endTime: string;
  isBooked: boolean;
  appointmentId?: string;
  patientName?: string;
  patientPhone?: string;
  status?: string;
  reason?: string;
}

export interface Schedule {
  id: string;
  doctorId: string;
  doctorName: string;
  specialtyName: string;
  workDate: string;
  startTime: string;
  endTime: string;
  slotDurationMinutes: number;
  totalSlots: number;
  availableSlots: number;
  isAvailable: boolean;
  note?: string;
  timeSlots: TimeSlot[];
}

export interface CreateScheduleDTO {
  doctorId?: string;
  workDate: string;
  startTime: string;
  endTime: string;
  slotDurationMinutes?: number;
  note?: string;
}

export interface UpdateScheduleDTO {
  note?: string;
  isAvailable?: boolean;
}

export interface ScheduleQueryParams {
  doctorId?: string;
  fromDate?: string;
  toDate?: string;
  page?: number;
  pageSize?: number;
  sortBy?: string;
  sortDir?: string;
}